### PR TITLE
clone view_context.config

### LINKED ISSUE #69: image_url and image_tag ignoring configured asset_host value
When I call `image_url("some_image.png")` directly in my views, I am getting a relative path vs a full asset url (which in our case is a CDN).

If I call `view_context.image_url("some_image.png")` it works fine and generates the full url.

Similar issues for `image_tag` which delegates to the same method.

I have temporarily added a `delegate :image_url, :image_tag, to: :view_context` in my component, but this feels wrong.

Is there something I am missing about how to configure this, or is this an actual bug?

### PR DESCRIPTION
**Pull request by mellowfish**
This is just a rough first attempt. Feedback on approach and style welcome. :)

Closes #69

### CODE REVIEW

**Comment by mellowfish** [comment]
@joelhawksley what do you think of this as a solution to issue #69?

**Comment by joelhawksley** [comment]
@mellowfish thanks for the ping. I owe you a reply!

It's a little suspicious to me to be generating these fairly specific methods on `Base`. I'm wondering if there might be a more general purpose solution here.

I think this might be directionally right: https://github.com/github/actionview-component/issues/69#issuecomment-542351438

As an aside, I'm curious what the use case is for having asset URLs in components. Might you be able to share more about the problem you're trying to solve?

**Comment by mellowfish** [comment]
@joelhawksley basically, if you want `image_tag`, `image_url`, etc to work when you have defined an `asset_host` URL. You might do so if you are uploading your assets to a CDN.

**Comment by mellowfish** [comment]
@joelhawksley also, what do you mean by `generating these fairly specific methods on Base`? I only added the one helper method, and all it is doing is copying a whitelisted set of values from view_context.config to `self.config`.

**Comment by joelhawksley** [comment]
> @joelhawksley also, what do you mean by `generating these fairly specific methods on Base`? I only added the one helper method, and all it is doing is copying a whitelisted set of values from `view_context.config` to `self.config`.

I'm concerned about having explicit cases for `asset_host` and `asset_dir`. Did you consider @cjbottaro's approach? I'm not even sure if it's feasible, but it feels more general purpose than these explicit cases.

**Comment by mellowfish** [comment]
@joelhawksley so, `self.class.instance_variable_set("@_config", view_context.config)` does appear to work, but that seems considerably more brittle than a whitelisted set of attributes to clone. For one thing, that means you can *never* configure your ActionView::Component class directly.

**Comment by mellowfish** [comment]
@joelhawksley what if I just copied over all the attributes, unless they were already defined locally? This would be similar to `local_config.merge!(view_context_config)`

**Comment by joelhawksley** [comment]
@mellowfish just wanted to let you know that we are looking at this PR internally and hope to have a response for you soon ❤️ . Thank you for your patience 🙏

File: test/app/components/asset_component.html.erb [hunk 1]
```diff
@@ -0,0 +1 @@
+<div><%= asset_url("application.css") %></div>
```
[the remark this answers is not in the document]
Comment (mellowfish) [reply test/app/components/asset_component.html.erb]: Hm. That does seem to work in the test, but I get an "undefined local variable or method `helpers'" when I try this in my actual app.

Also, while that is a reasonable patch, so is my current `delegate :image_url, :image_path, to: :view_context`. I was hoping for something that wouldn't require so much indirection in the consuming components

[the remark this answers is not in the document]
Comment (joelhawksley) [reply test/app/components/asset_component.html.erb]: @mellowfish which version are you using in your app? 

We've gone with the `helpers` approach to match how Rails provides access to view helpers in controllers. This behavior makes it clear that  the component is accessing state / methods outside the scope of the component.

[the remark this answers is not in the document]
Comment (mellowfish) [reply test/app/components/asset_component.html.erb]: Hm, let me update and try again

[the remark this answers is not in the document]
Comment (mellowfish) [reply test/app/components/asset_component.html.erb]: Well, having other troubles after the update, likely due to #79, though I my namespace is `ModuleA::ModuleB::ActivityC`, with no duplicate names.

**Comment by mellowfish** [comment]
I guess if `helpers` is the general strategy for the future that is ok.

**Comment by joelhawksley** [comment]
@mellowfish I'd love to still land this PR with the test cases you wrote, so that we can make sure that `helpers.asset_url` continues to work ❤️

**Comment by mellowfish** [comment]
Ok, removing "fix" and just doing the test

**Pull request closed without merging**